pkg/cluster/config/types.go

```diff
@@ -54,6 +56,9 @@ type Node struct {
 	// KubeadmConfigPatchesJSON6902 are applied to the generated kubeadm config
 	// as patchesJson6902 to `kustomize build`
 	KubeadmConfigPatchesJSON6902 []kustomize.PatchJSON6902
+	// Mounts describes additional mount points for the node container
+	// These may be used to EG bind a hostpath
+	Mounts []cri.Mount `json:"mounts,omitempty"`
```

Comment:
there is a weird side effect when marshaling slices with omitempty, that is already a problem in k/k, kubeadm, cluster-api. i think the safe route was to use `Mounts *[]cri.Mount `json:"mounts,omitempty"` but i need to check this.

Reply:
```
package main

import (
	"fmt"
	"log"
	"sigs.k8s.io/yaml"
)

type A struct {
	Mounts []string `json:"mounts,omitempty"`
}

func main() {
	a := A{Mounts: []string{}} // <--------- [*]
	fmt.Println(a.Mounts, a.Mounts == nil)
	b, err := yaml.Marshal(a)	
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(string(b))
	c := A{}
	err = yaml.Unmarshal(b, &c)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(c.Mounts, c.Mounts == nil)
}
```

```
[] false
{}
[] true
```

[*]
so this problem is relevant only if we assign a meaning to the "empty" value of `Mounts` and what is the "empty" value (is it a empty slice or `nil`) and if the user is marhaling the kind Config.

if there is no default value this problem is not relevant, but just referencing this here so that we don't have this problem for the future kind config:
https://github.com/kubernetes/kubeadm/issues/1358

Reply:
What was it? https://play.golang.org/p/EOfDnWThpdj

Reply:
I _think_ we can treat empty and nil the same, the node already has mounts (tmps and one docker volume) we're just adding additional mounts here (we should clarify that in the docs)

Reply:
hm, yes. i know the existing mounts are essential for the setup to work.

so in a way we already have default mounts and these mounts are extra mounts :thinking: 
would it be clearer naming this `ExtraMounts` and including some more nodes in the field comment?

Reply:
Yeah I think that might make sense.

Reply:
renamed it 👍